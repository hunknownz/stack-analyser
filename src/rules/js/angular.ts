import { register } from '../../rules.js';

register({
  tech: 'angular',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
