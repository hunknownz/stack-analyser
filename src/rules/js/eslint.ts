import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: [
    '.eslintrc',
    '.eslintrc.cjs',
    '.eslintrc.json',
    '.eslintrc.js',
    'eslint.config.js',
    'eslint.config.cjs',
    'eslint.config.mjs',
  ],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
