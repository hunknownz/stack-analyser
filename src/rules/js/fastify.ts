import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'fastify' }],
});
