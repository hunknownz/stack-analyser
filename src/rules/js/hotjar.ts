import { register } from '../../register.js';

register({
  tech: 'hotjar',
  name: 'Hotjar',
  type: 'tool',
  dependencies: [{ type: 'npm', name: '@hotjar/browser' }],
});
