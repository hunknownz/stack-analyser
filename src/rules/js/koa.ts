import { register } from '../../register.js';

register({
  tech: 'koa',
  name: 'Koa',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'koa' }],
});
