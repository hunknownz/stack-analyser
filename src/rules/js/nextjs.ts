import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'framework',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
