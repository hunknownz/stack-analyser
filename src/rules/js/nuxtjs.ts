import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'framework',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
