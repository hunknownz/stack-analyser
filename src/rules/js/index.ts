import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
