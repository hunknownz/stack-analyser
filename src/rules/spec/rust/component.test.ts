import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';
import '../../../autoload.js';

describe('rust (component)', () => {
  it('should match everything', async () => {
    const lockfile: string[] = [
      `[package]
name = "svix-server"
version = "1.8.0"

[dependencies]
`,
    ];

    for (const item of rawList) {
      if (item.type !== 'dependency' || item.ref.type !== 'rust') {
        continue;
      }

      const example = 'example' in item.ref ? item.ref.example : item.ref.name;
      lockfile.push(`${example} = "1.0.0"\n`);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['Cargo.toml'],
        },
        files: {
          '/Cargo.toml': lockfile.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });

    expect([...merged.techs].sort()).toMatchSnapshot();
    expect([...merged.dependencies].sort()).toMatchSnapshot();
  });

  it('should match foreign dependencies', async () => {
    const lockfile: string[] = [
      `[package]
name = "svix-server"
version = "1.8.0"

[dependencies]
test = { path = "../test" }
hello = { path = "../world", version = "1.0.0" }
axum = { version = "0.6.1", features = ["headers"] }

[dev-dependencies]
anyhow = "1.0.56"
redis_cluster_async = { git = "https://github.com/redis-rs/redis-cluster-async.git", rev = "e6fe168" }

[build-dependencies]
foobar = "^1.0.0"
`,
    ];

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['Cargo.toml'],
        },
        files: {
          '/Cargo.toml': lockfile.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });

    expect([...merged.dependencies].sort()).toMatchSnapshot();
  });

  it('should match license', async () => {
    const lockfile: string[] = [
      `[package]
name = "svix-server"
version = "1.8.0"
license = "Apache-2.0"
`,
    ];

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['Cargo.toml'],
        },
        files: {
          '/Cargo.toml': lockfile.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });

    expect([...merged.licenses.values()]).toStrictEqual(['Apache-2.0']);
  });
});
