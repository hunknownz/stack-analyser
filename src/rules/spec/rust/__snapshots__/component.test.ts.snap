// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rust (component) > should match everything 1`] = `
Set {
  "zookeeper",
  "aws",
  "aws.dynamodb",
  "cassandra",
  "couchbase",
  "elasticsearch",
  "gcp",
  "gcp.bigquery",
  "influxdb",
  "memcached",
  "mongodb",
  "mysql",
  "postgresql",
  "redis",
  "sqlite",
  "snowflake",
  "hashicorp_vault",
  "fastly",
  "rabbitmq",
  "gcp.pubsub",
  "algolia",
  "auth0",
  "datadog",
  "mailjet",
  "okta",
  "slack",
  "sentry",
  "neo4j",
  "gitlab",
  "pagerduty",
  "rust",
}
`;

exports[`rust (component) > should match everything 2`] = `
[
  [
    "rust",
    "algoliasearch",
    "1.0.0",
  ],
  [
    "rust",
    "auth0",
    "1.0.0",
  ],
  [
    "rust",
    "aws-sdk-dynamodb",
    "1.0.0",
  ],
  [
    "rust",
    "aws-sdk-elasticsearch",
    "1.0.0",
  ],
  [
    "rust",
    "cassandra-cpp",
    "1.0.0",
  ],
  [
    "rust",
    "couchbase",
    "1.0.0",
  ],
  [
    "rust",
    "datadog-apm",
    "1.0.0",
  ],
  [
    "rust",
    "ddtrace",
    "1.0.0",
  ],
  [
    "rust",
    "discord-cassandra-cpp",
    "1.0.0",
  ],
  [
    "rust",
    "elasticsearch",
    "1.0.0",
  ],
  [
    "rust",
    "fastly-api",
    "1.0.0",
  ],
  [
    "rust",
    "gcp-bigquery-client",
    "1.0.0",
  ],
  [
    "rust",
    "gitlab",
    "1.0.0",
  ],
  [
    "rust",
    "google-cloud-pubsub",
    "1.0.0",
  ],
  [
    "rust",
    "hashicorp_vault",
    "1.0.0",
  ],
  [
    "rust",
    "influxdb",
    "1.0.0",
  ],
  [
    "rust",
    "influxdb2",
    "1.0.0",
  ],
  [
    "rust",
    "libsqlite3-sys",
    "1.0.0",
  ],
  [
    "rust",
    "mailjet-rs",
    "1.0.0",
  ],
  [
    "rust",
    "memcache",
    "1.0.0",
  ],
  [
    "rust",
    "memcache-async",
    "1.0.0",
  ],
  [
    "rust",
    "mongodb",
    "1.0.0",
  ],
  [
    "rust",
    "mysql_async",
    "1.0.0",
  ],
  [
    "rust",
    "neo4rs",
    "1.0.0",
  ],
  [
    "rust",
    "okta",
    "1.0.0",
  ],
  [
    "rust",
    "pagerduty-rs",
    "1.0.0",
  ],
  [
    "rust",
    "postgres",
    "1.0.0",
  ],
  [
    "rust",
    "rabbitmq-stream-client",
    "1.0.0",
  ],
  [
    "rust",
    "redis",
    "1.0.0",
  ],
  [
    "rust",
    "redis-async",
    "1.0.0",
  ],
  [
    "rust",
    "rs-snowflake",
    "1.0.0",
  ],
  [
    "rust",
    "rusqlite",
    "1.0.0",
  ],
  [
    "rust",
    "sentry",
    "1.0.0",
  ],
  [
    "rust",
    "slack-morphism",
    "1.0.0",
  ],
  [
    "rust",
    "snowflake",
    "1.0.0",
  ],
  [
    "rust",
    "sqlite",
    "1.0.0",
  ],
  [
    "rust",
    "zookeeper",
    "1.0.0",
  ],
]
`;

exports[`rust (component) > should match foreign dependencies 1`] = `
[
  [
    "rust",
    "anyhow",
    "1.0.56",
  ],
  [
    "rust",
    "axum",
    "0.6.1",
  ],
  [
    "rust",
    "foobar",
    "^1.0.0",
  ],
  [
    "rust",
    "hello",
    "path:../world:1.0.0",
  ],
  [
    "rust",
    "redis_cluster_async",
    "git:https://github.com/redis-rs/redis-cluster-async.git#e6fe168",
  ],
  [
    "rust",
    "test",
    "path:../test",
  ],
]
`;
